### PR TITLE
fix discovery plugin specs

### PR DESCRIPTION
**Pull request by mmoll**

### CODE REVIEW

File: spec/classes/foreman_proxy__plugin__discovery_spec.rb [hunk 1]
```diff
@@ -4,12 +4,19 @@
   on_supported_os.each do |os, facts|
     let(:facts) { facts }
```
Comment (domcleal) [inline spec/classes/foreman_proxy__plugin__discovery_spec.rb]: This needs to move inside the OS context block, it should then fix the intermittent failures (https://github.com/theforeman/puppet-foreman/pull/377/files#r46414771).  The case statement will also need to handle all supported OSes.

**Comment by ekohl** [comment]
:+1: on green tests

**Comment by mmoll** [comment]
:green_heart:

**Comment by ekohl** [comment]
Thanks!

**Pull request merged**